x-pack/plugins/security_solution/public/management/pages/policy/store/policy_details/selectors.ts

```diff
@@ -117,17 +121,32 @@ export const fullPolicy: (s: Immutable<PolicyDetailsState>) => PolicyConfig = cr
   }
 );
 
+/**
+ * Returns the full Endpoint Policy, with any paid/licensed features that are not
+ * currently allowed for the current license level reset to defaults.
+ */
+const policyStrippedUnlicensedFeatures: (s: PolicyDetailsState) => PolicyConfig = createSelector(
+  fullPolicy,
+  licenseState,
+  (policyData, license) => {
+    return unsetPolicyFeaturesAboveLicenseLevel(policyData, license as ILicense);
```

Comment:
Surprised you have to cast `license`. was it because it could be undefined? if so, I would suggest you instead used `!` which remove the undefined from the type.

Reply:
this was not about being `undefined` or `null`. But because `licenseState` gives an `ImmutableObject<ILicense>`. Since it's pulling from a store.

But the `unset..` function wants a normal `ILicense`, because that's a normal type. It doesn't modify the license so I just cast it.